### ISSUE TITLE
defcase doesn't work with setup args

### ISSUE THREAD

**Issue by akhudek** [davidsantiago/perforate#4]
defcase doesn't appear to work with setup arguments as described in the docs. 

e.g. 

``` clojure
(defgoal simple "Simple." :setup (fn [] [1 2]))

(defcase simple :default [a b] (+ a b))
```

**Comment by davidsantiago** [comment]
Whoops, you're right. When I first saw this, I coulda sworn defcase worked right, but I realized I've only been using it without a :setup function. The problem is that "[]" and "~arg-list" are swapped here: https://github.com/davidsantiago/perforate/blob/master/src/perforate/core.clj#L86

**Comment by davidsantiago** [comment]
Should be fixed in version 0.3.1. Thanks for the issue report.

**Comment by akhudek** [comment]
Thanks for the fix. Love the library, it's been very useful.
